### PR TITLE
[TimeVaryingDCMPlanner] restructure of setDCMReference() and setContactPhaseList()

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
#208 implements   `setDCMReference()` however the user had to call the function after `setContactPhaseList()`.  Thanks to this PR, `setDCMReference()` does not depends on `setContactPhaseList()` and the user can call it whenever their want. 

cc @paolo-viceconte

### CODE REVIEW

**Review by prashanthr05 — APPROVED** [review]
Ok for me! But I would wait for @S-Dafarra approval as well.

File: src/Planners/src/TimeVaryingDCMPlanner.cpp
Comment (S-Dafarra) [inline src/Planners/src/TimeVaryingDCMPlanner.cpp:497]: Maybe we can add some tags to distinguish between errors and info messages.

**Review by S-Dafarra — APPROVED** [review]
Just a simple suggestion. So basically, you can set the initial dcm condition without changing the foot list, right?

**Comment by GiulioRomualdi** [comment]
>  So basically, you can set the initial dcm condition without changing the foot list, right?

Right

**Comment by GiulioRomualdi** [comment]
Comment https://github.com/dic-iit/bipedal-locomotion-framework/pull/220#issuecomment-789555938 updated. @S-Dafarra

File: src/Planners/src/TimeVaryingDCMPlanner.cpp
Comment (GiulioRomualdi) [reply src/Planners/src/TimeVaryingDCMPlanner.cpp:497]: I would do it in a separate PR for all the error/warning/info in the repo. Indeed now all the errors/warnings/info are printed with `cerr` and there is not a common function to call.  https://github.com/dic-iit/bipedal-locomotion-framework/issues/137#issuecomment-736655938

Comment (GiulioRomualdi) [reply src/Planners/src/TimeVaryingDCMPlanner.cpp:497]: For instance, we can use something like this: https://github.com/robotology/idyntree/blob/2e5cf7180a517bf91777d48761c4c12c94e7e404/src/core/include/iDynTree/Core/Utils.h#L66-L93

Comment (S-Dafarra) [reply src/Planners/src/TimeVaryingDCMPlanner.cpp:497]: I was not referring to ``cerr`` but rather to the message itself. It is not clear whether the message is an error or not. I was suggesting adding something like ``[ERROR]`` since it is going to be much easier to find them in the middle of tons of other messages.

Comment (GiulioRomualdi) [reply src/Planners/src/TimeVaryingDCMPlanner.cpp]: On that one I agree done in https://github.com/dic-iit/bipedal-locomotion-framework/pull/220/commits/7999c2f0010291c9bda996d3a3b0eb218ce7c988

**Pull request merged**
